testing/checkpoints/test_checkpoint_pipeline_runner.py

```diff
@@ -0,0 +1,134 @@
+chekpoint_not_saved = SomeCheckpointStep(None, None)
+tape = TapeCallbackFunction()
+
+tape_without_checkpoint_test_arguments = (
+    [TransformCallbackStep(tape.callback, ["1"]),
+     TransformCallbackStep(tape.callback, ["2"]),
+     TransformCallbackStep(tape.callback, ["3"])],
+    ["1", "2", "3"])
+
+tape_checkpoint_not_saved_test_arguments = (
+    [TransformCallbackStep(tape.callback, ["1"]),
+     SomeCheckpointStep(None, None),
+     TransformCallbackStep(tape.callback, ["2"]),
+     TransformCallbackStep(tape.callback, ["3"])],
+    ["1", "2", "3"])
+
+tape_checkpoint_saved_after_first_step_test_arguments = (
+    [TransformCallbackStep(tape.callback, ["1"]),
+     SomeCheckpointStep(data_inputs, expected_outputs),
+     TransformCallbackStep(tape.callback, ["2"]),
+     TransformCallbackStep(tape.callback, ["3"])],
+    ["2", "3"])
+
+tape_checkpoint_saved_after_second_step_test_arguments = (
+    [TransformCallbackStep(tape.callback, ["1"]),
+     TransformCallbackStep(tape.callback, ["2"]),
+     SomeCheckpointStep(data_inputs, expected_outputs),
+     TransformCallbackStep(tape.callback, ["3"])],
+    ["3"])
+
+tape_checkpoint_saved_after_last_step_test_arguments = (
+    [TransformCallbackStep(tape.callback, ["1"]),
+     TransformCallbackStep(tape.callback, ["2"]),
+     TransformCallbackStep(tape.callback, ["3"]),
+     SomeCheckpointStep(data_inputs, expected_outputs), ],
+    [])
+
+
+@pytest.mark.parametrize("steps,expected_tape", [
+    tape_without_checkpoint_test_arguments,
+    tape_checkpoint_not_saved_test_arguments,
+    tape_checkpoint_saved_after_first_step_test_arguments,
+    tape_checkpoint_saved_after_second_step_test_arguments,
+    tape_checkpoint_saved_after_last_step_test_arguments,
+])
+def test_fit_transform(steps, expected_tape):
+    tape.data = []
+    tape.name_tape = []
+    pipeline = Pipeline(
+        steps=steps,
+        pipeline_runner=CheckpointPipelineRunner()
+    )
+
+    actual_pipeline, actual_data_inputs = pipeline.fit_transform(data_inputs, expected_outputs)
+
+    actual_tape = tape.get_name_tape()
+    assert isinstance(actual_pipeline, Pipeline)
+    assert actual_tape == expected_tape
+    assert np.array_equal(actual_data_inputs, data_inputs)
+
+
+@pytest.mark.parametrize("steps,expected_tape", [
+    tape_without_checkpoint_test_arguments,
+    tape_checkpoint_not_saved_test_arguments,
+    tape_checkpoint_saved_after_first_step_test_arguments,
+    tape_checkpoint_saved_after_second_step_test_arguments,
+    tape_checkpoint_saved_after_last_step_test_arguments,
+])
+def test_should_fit_each_steps(steps, expected_tape):
+    tape.data = []
+    tape.name_tape = []
+    pipeline = Pipeline(
+        steps=steps,
+        pipeline_runner=CheckpointPipelineRunner()
+    )
+
+    actual_pipeline = pipeline.fit(data_inputs, expected_outputs)
+
+    actual_tape = tape.get_name_tape()
+    assert isinstance(actual_pipeline, Pipeline)
+    assert actual_tape == expected_tape
```

Comment:
Does this really works? It seems to me that you used a `TransformCallbackStep` and so that it might not be called on the `Fit`s. Perhaps that the Runner runs a transform after every fit to pass data to the next step to fit... If so, it's still okay, although it'd be best to check the actual "fit".